include/zephyr/linker/irq-vector-table-section.ld

```diff
@@ -0,0 +1,4 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+. = ALIGN(CONFIG_ARCH_IRQ_VECTOR_TABLE_ALIGN);
```

Comment:
Please correct me, but are we sure that this change isn't introduce regression? Previously we align _IRQ_VECTOR_TABLE_SECTION_SYMS with default section alignment (at least), but now we have `CONFIG_ARCH_IRQ_VECTOR_TABLE_ALIGN` with default value '0' - so now we have
`. = ALIGN(0);`

Not fully sure what linker will do with `ALIGN(0)`

Reply:
No one is fully sure what a linker is going to do with anything.  But I'd be very surprised if this blew up.

Reply:
It doesn't sound too complex to find out what happens. I assume for gcc we have CI coverage. Evgeniy, your concern is about the llvm linker?

Reply:
@evgeniy-paltsev
> Please correct me, but are we sure that this change isn't introduce regression? 

"the only certainty in life is death and taxes" for all the rest there is CI, and CI is all green. If a new arch needs some special alignment (like RISCV), then the newly introduced `CONFIG_ARCH_IRQ_VECTOR_TABLE_ALIGN` can be used.

For all the other architectures FWICT the alignment of the generated IRQ vector table is not important (please note that this vector table is not the same as https://github.com/zephyrproject-rtos/zephyr/blob/main/arch/arm/core/aarch32/cortex_m/vector_table.S, just to avoid confusion)

> Previously we align _IRQ_VECTOR_TABLE_SECTION_SYMS with default section alignment (at least)

For ARM and ARC the vector table is placed after `.exc_vector_table.*` symbols, so it is really not aligned to anything defined (and this is one reason indeed to introduce `CONFIG_ARCH_IRQ_VECTOR_TABLE_ALIGN`)

Also please consider that this PR is not touching the address where the `_irq_vector_table` is placed: the `vector_table.ld` file is placed with key `0x0vectors`, the same sort key is used to place the `irq-vector-table-section.ld`, so nothing is changing (as long as there are no `.vectors` symbols, and there are not).

> Not fully sure what linker will do with ALIGN(0)

I'm not inventing anything new here, this is already used in Zephyr, see https://github.com/zephyrproject-rtos/zephyr/blob/f79eaba7e6d2c698fbaba3f7bc7348f9a78a611d/arch/Kconfig#L421-L423
https://github.com/zephyrproject-rtos/zephyr/blob/f79eaba7e6d2c698fbaba3f7bc7348f9a78a611d/include/zephyr/linker/common-ram.ld#L25-L26
so I would assume this is safe.

Reply:
Ok, I've run verification with MWDT toolchain, though maybe it's better to set CONFIG_ARCH_SW_ISR_TABLE_ALIGN to 1 by default instead of 0?

So we get 
```
. = ALIGN(1); 
```
instead of
```
. = ALIGN(0); 
```
by default. At least it's more meaningful and I expect it should be more acceptable for all linkers.

@carlocaione  What do you think?

Reply:
I think that at this point we should set a more proper value for both `ARCH_IRQ_VECTOR_TABLE_ALIGN` and `ARCH_SW_ISR_TABLE_ALIGN` that it should probably be `4` to have a 32-bit alignment.